hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/KeyPathRetriever.java

```diff
@@ -0,0 +1,295 @@
+ */
+@CommandLine.Command(
+    name = "retrieve-key-fullpath",
+    description = "retrieve full key path with matching criteria")
+@MetaInfServices(SubcommandWithParent.class)
+public class KeyPathRetriever implements Callable<Void>, SubcommandWithParent {
+  private static final DBColumnFamilyDefinition<Long, String> DIRTREE_COLUMN_FAMILY
+      = new DBColumnFamilyDefinition<>("dirTreeTable", LongCodec.get(), StringCodec.get());
+
+  @CommandLine.Spec
+  private static CommandLine.Model.CommandSpec spec;
+
+  @CommandLine.Option(names = {"--db"},
+      required = true,
+      description = "Database File")
+  private String dbFile;
+
+  @CommandLine.Option(names = {"--containers"},
+      required = false,
+      description = "Comma separated Container Ids")
+  private String strContainerIds;
+
+  @CommandLine.Option(names = {"--container-file"},
+      required = false,
+      description = "file with container id in new line")
+  private String containerFileName;
+
+  @CommandLine.Option(names = {"--out", "-o"},
+      required = false,
+      description = "out file name")
+  private String fileName;
+
+  private DBStore dirTreeDbStore = null;
+  private Table<Long, String> dirTreeTable = null;
+
+  @Override
+  public Class<?> getParentType() {
+    return OzoneDebug.class;
+  }
+
+  public void setDirTreeTable(Table<Long, String> table) {
+    this.dirTreeTable = table;
+  }
+
+  @Override
+  public Void call() throws Exception {
+    // get containerIds filter
+    Set<Long> containerIds = new HashSet<>();
+    if (!StringUtils.isEmpty(strContainerIds)) {
+      if (!StringUtils.isEmpty(strContainerIds)) {
+        String[] split = strContainerIds.split(",");
+        for (String id : split) {
+          containerIds.add(Long.parseLong(id));
+        }
+      }
+    } else if (!StringUtils.isEmpty(containerFileName)) {
+      try (Stream<String> stream = Files.lines(Paths.get(containerFileName))) {
+        stream.forEach(e -> containerIds.add(Long.parseLong(e)));
+      }
+    }
+    if (containerIds.isEmpty()) {
+      System.out.println("No containers to be filtered");
+      return null;
+    }
+    // out stream
+    PrintWriter writer;
+    if (StringUtils.isEmpty(fileName)) {
+      writer = out();
+    } else {
+      writer = new PrintWriter(new BufferedWriter(new PrintWriter(fileName, UTF_8.name())));
+    }
+
+    // db handler
+    OMMetadataManager metadataManager = getOmMetadataManager(dbFile);
+    if (metadataManager == null) {
+      writer.close();
+      return null;
+    }
+
+    dirTreeDbStore = openDb(new File(dbFile).getParentFile());
+    if (dirTreeDbStore == null) {
+      writer.close();
+      metadataManager.stop();
+      return null;
+    }
+    dirTreeTable = dirTreeDbStore.getTable(DIRTREE_COLUMN_FAMILY.getName(), Long.class, String.class);
+
+    try {
+      retrieve(metadataManager, writer, containerIds);
+    } finally {
+      dirTreeDbStore.close();
```

Comment:
Temporary db can be deleted as well, as it not used again.

Reply:
@ashishkumar50 done